src/shared/components/ui/Toast.tsx

```
@@ -0,0 +1,73 @@
+
+export const Toast = ({
+  message,
+  onClose,
+  variant = "info",
+  duration = 3000,
+  className,
+}: ToastProps) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const onCloseRef = useRef(onClose);
+
+  onCloseRef.current = onClose;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/shared/components/ui/Toast.tsx:25:3
  23 |   const onCloseRef = useRef(onClose);
  24 |
> 25 |   onCloseRef.current = onClose;
     |   ^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  26 |
  27 |   useEffect(() => {
  28 |     const frameId = window.requestAnimationFrame(() => {

(react-hooks/refs)